CDC-Data-Reconciliation-Backend/compare.py

```diff
@@ -109,6 +109,13 @@ def comp(state_dict, cdc_dict):
                 if i < 1:
                     i += 1
                     continue
+                
+                # Determine which attributes to compare: specified ones or all
+                attributes_to_compare = compare_attributes if compare_attributes else state_row.keys()
+                for attribute in attributes_to_compare:
+                    # Skip if the attribute is not in the CDC dict
+                    if attribute not in cdc_dict[state_case_id]:
```

Comment:
All attributes that are passed in should be in the cdc dataset. I believe you should be editing the for loop above which loops over all attributes. Line 107 is what I'm referencing. Should loop over all attributes that were passed in or loop over all the state attributes if none were passed in via cli arguments.